### PR TITLE
feat(table): add variant input to table

### PR DESCRIPTION
**Pull request by elenagarrone**
# Description

Add a new `variant` input to the table to cater for a new alternative style. This style will be mainly used to overcome accessibility issues when the text is a lighter colour (for example with the blue links).

## Requirements

Please briefly outline any requirements for the work which may aid the testing and review process.

Storybook link: https://deploy-preview-155--legal-and-general-canopy.netlify.app/?path=/story/components-table--standard
Design link: (link to design or delete if not required)
Screenshots:
Expandable table:
Screenshot 2021-01-12 at 17 01 39
Screenshot 2021-01-12 at 17 01 53

Regular table:
Screenshot 2021-01-12 at 17 02 23
Screenshot 2021-01-12 at 17 02 12

# Checklist:

- [x] The commit messages follow the [convention for this project](./blob/master/docs/CONTRIBUTING.md#conventional-commits)
- [x] I have provided an adequate amount of test coverage
- [ ] I have added the functionality to the [test app](./blob/master/docs/CONTRIBUTING.md#build-test-application)
- [x] I have provided a story in storybook to document the changes
- [x] I have provided documentation in the notes section of the story
- [ ] I have added any new public feature modules to public-api.ts
- [ ] I have [linked the new component](<(https://github.com/Legal-and-General/canopy/blob/master/docs/CONTRIBUTING.md#invision-dsm)>) to adobe DSM (if appropriate)

### CODE REVIEW

File: projects/canopy/src/lib/table/table/table.component.spec.ts [hunk 1]
```diff
@@ -84,12 +84,23 @@ describe('TableComponent', () => {
...
   });
 
+  describe('when a variant is specified', () => {
+    it('should set the correct class modifier', () => {
+      component.variant = 'bordered';
+      fixture.detectChanges();
+
+      expect(tableDebugElement.nativeElement.getAttribute('class')).toContain(
+        'lg-table--bordered',
+      );
```
Comment (owensgit) [inline projects/canopy/src/lib/table/table/table.component.spec.ts:100]: Perhaps we should also test here that the doesn't have the `lg-table--striped` class, as there could be a risk that the component logic fails to remove one while adding another. The could go for the `it('should have the default classes')` test above, perhaps that check that the bordered class is not present.

**Review by owensgit — CHANGES_REQUESTED** [review]
Nice addition @elenagarrone :) Just a couple of minor comments from me.

File: package.json [hunk 2]
```diff
@@ -144,7 +144,7 @@
     ],
     "projects/canopy/src/**/*.{scss, css}": [
       "prettier --write",
-      "stylelint --fix",
+      "stylelint --syntax scss --fix",
```
[the remark this answers is not in the document]
Comment (elenagarrone) [reply package.json:147]: Yes, i was unsure about it but it does make sense for it to be in a separate commit 👍

File: projects/canopy/src/lib/table/table/table.component.spec.ts (hunk 1 above)
Comment (elenagarrone) [reply projects/canopy/src/lib/table/table/table.component.spec.ts:103]: Done

File: projects/canopy/src/lib/table/table-row/table-row.component.ts [hunk 3]
```diff
@@ -46,9 +47,31 @@ export class LgTableRowComponent {
...
+    }
+    this._tableCellElement = element;
+  }
+  get tableCellElement(): LgTableCellComponent {
+    return this._tableCellElement;
+  }
+
+  @HostBinding('class.lg-table-row__toggle--active')
+  get isToggledActive(): boolean {
+    return this.tableCellComponent?.toggleClass?.isActive;
```
[the remark this answers is not in the document]
Comment (elenagarrone) [reply projects/canopy/src/lib/table/table-row/table-row.component.ts]: Done 👍

File: projects/canopy/src/lib/table/table-row/table-row.component.spec.ts [hunk 4]
```diff
@@ -36,6 +39,40 @@ describe('LgTableRowComponent', () => {
...
+  it('should have the active class if the row is toggled active', () => {
+    when(componentSpy.isToggledActive).thenReturn(true);
+    fixture.detectChanges();
+
+    expect(fixture.nativeElement.getAttribute('class')).toContain(
+      'lg-table-row__toggle--active',
+    );
+  });
+
+  it('should have the active class if the row is not toggled active', () => {
```
[the remark this answers is not in the document]
Comment (elenagarrone) [reply projects/canopy/src/lib/table/table-row/table-row.component.spec.ts]: Done

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (elenagarrone) [reply package.json:147]: Done

**Review by pauleustice — APPROVED** [review]

**Review by owensgit — DISMISSED** [review]

File: projects/canopy/src/lib/table/table-row/table-row.component.ts [hunk 5]
```diff
@@ -46,9 +47,31 @@ export class LgTableRowComponent {
 
   @ContentChildren(LgTableCellComponent) bodyCells: QueryList<LgTableCellComponent>;
 
-  @ContentChildren(LgTableHeadCellComponent) headCells: QueryList<
-    LgTableHeadCellComponent
-  >;
+  @ContentChildren(LgTableHeadCellComponent)
+  headCells: QueryList<LgTableHeadCellComponent>;
+
+  private _tableCellElement: LgTableCellComponent;
```
Comment (thatguynamedandy) [inline projects/canopy/src/lib/table/table-row/table-row.component.ts:53]: I'm slightly confused by the `tableCellElement` whereabouts is it used? It was initially the naming that confused me with `tableCellElement` and `tableCellComponent` but both being of type `LgTableCellComponent`.

Comment (elenagarrone) [reply projects/canopy/src/lib/table/table-row/table-row.component.ts]: Apologies that was a leftover bit of code! I have remove it now as unnecessary

File: projects/canopy/src/lib/table/table/table.component.scss [hunk 6]
```diff
@@ -7,24 +7,34 @@
...
+      border-bottom: solid var(--border-width) var(--table-row-border-color);
     }
   }
 
-  &:not(&--expandable) {
-    tr {
-      &:nth-child(even) {
-        background-color: var(--table-stripe-color);
+  &--expandable {
+    &.lg-table--striped {
```
[the remark this answers is not in the document]
Comment (elenagarrone) [reply projects/canopy/src/lib/table/table/table.component.scss:23]: I guess I could have moved `&.lg-table--striped {` and `tr` on the same line. I can do that if you want ;)

**Review by thatguynamedandy — APPROVED** [review]

File: projects/canopy/src/lib/table/table/table.component.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (thatguynamedandy) [reply projects/canopy/src/lib/table/table/table.component.scss:23]: nah I think it's good, it's more a wider concern that too many table options might get tricky, but I don't think we are there yet.

**Review by owensgit — APPROVED** [review]

**Pull request merged**